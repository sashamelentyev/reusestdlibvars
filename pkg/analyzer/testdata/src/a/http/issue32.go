// Code generated by usestdlibvars, DO NOT EDIT.

package http_test

import "net/http"

var (
	_ = "connect"
	_ = "delete"
	_ = "get"
	_ = "head"
	_ = "options"
	_ = "patch"
	_ = "post"
	_ = "put"
	_ = "trace"
)

const (
	_ = "connect"
	_ = "delete"
	_ = "get"
	_ = "head"
	_ = "options"
	_ = "patch"
	_ = "post"
	_ = "put"
	_ = "trace"
)

func _() {
	_, _ = http.NewRequest("connect", "", http.NoBody) // want `"CONNECT" can be replaced by http\.MethodConnect`
	_, _ = http.NewRequest("delete", "", http.NoBody)  // want `"DELETE" can be replaced by http\.MethodDelete`
	_, _ = http.NewRequest("get", "", http.NoBody)     // want `"GET" can be replaced by http\.MethodGet`
	_, _ = http.NewRequest("head", "", http.NoBody)    // want `"HEAD" can be replaced by http\.MethodHead`
	_, _ = http.NewRequest("options", "", http.NoBody) // want `"OPTIONS" can be replaced by http\.MethodOptions`
	_, _ = http.NewRequest("patch", "", http.NoBody)   // want `"PATCH" can be replaced by http\.MethodPatch`
	_, _ = http.NewRequest("post", "", http.NoBody)    // want `"POST" can be replaced by http\.MethodPost`
	_, _ = http.NewRequest("put", "", http.NoBody)     // want `"PUT" can be replaced by http\.MethodPut`
	_, _ = http.NewRequest("trace", "", http.NoBody)   // want `"TRACE" can be replaced by http\.MethodTrace`
}

func _() {
	_, _ = http.NewRequestWithContext(nil, "connect", "", http.NoBody) // want `"CONNECT" can be replaced by http\.MethodConnect`
	_, _ = http.NewRequestWithContext(nil, "delete", "", http.NoBody)  // want `"DELETE" can be replaced by http\.MethodDelete`
	_, _ = http.NewRequestWithContext(nil, "get", "", http.NoBody)     // want `"GET" can be replaced by http\.MethodGet`
	_, _ = http.NewRequestWithContext(nil, "head", "", http.NoBody)    // want `"HEAD" can be replaced by http\.MethodHead`
	_, _ = http.NewRequestWithContext(nil, "options", "", http.NoBody) // want `"OPTIONS" can be replaced by http\.MethodOptions`
	_, _ = http.NewRequestWithContext(nil, "patch", "", http.NoBody)   // want `"PATCH" can be replaced by http\.MethodPatch`
	_, _ = http.NewRequestWithContext(nil, "post", "", http.NoBody)    // want `"POST" can be replaced by http\.MethodPost`
	_, _ = http.NewRequestWithContext(nil, "put", "", http.NoBody)     // want `"PUT" can be replaced by http\.MethodPut`
	_, _ = http.NewRequestWithContext(nil, "trace", "", http.NoBody)   // want `"TRACE" can be replaced by http\.MethodTrace`
}

func _() {
	_ = &http.Request{
		Method: "connect", // want `"CONNECT" can be replaced by http\.MethodConnect`
	}
	_ = &http.Request{
		Method: "delete", // want `"DELETE" can be replaced by http\.MethodDelete`
	}
	_ = &http.Request{
		Method: "get", // want `"GET" can be replaced by http\.MethodGet`
	}
	_ = &http.Request{
		Method: "head", // want `"HEAD" can be replaced by http\.MethodHead`
	}
	_ = &http.Request{
		Method: "options", // want `"OPTIONS" can be replaced by http\.MethodOptions`
	}
	_ = &http.Request{
		Method: "patch", // want `"PATCH" can be replaced by http\.MethodPatch`
	}
	_ = &http.Request{
		Method: "post", // want `"POST" can be replaced by http\.MethodPost`
	}
	_ = &http.Request{
		Method: "put", // want `"PUT" can be replaced by http\.MethodPut`
	}
	_ = &http.Request{
		Method: "trace", // want `"TRACE" can be replaced by http\.MethodTrace`
	}
}
