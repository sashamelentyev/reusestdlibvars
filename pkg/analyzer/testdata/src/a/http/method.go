// Code generated by usestdlibvars, DO NOT EDIT.

package http_test

import "net/http"

var (
	_ = "CONNECT"
	_ = "DELETE"
	_ = "GET"
	_ = "HEAD"
	_ = "OPTIONS"
	_ = "PATCH"
	_ = "POST"
	_ = "PUT"
	_ = "TRACE"
)

const (
	_ = "CONNECT"
	_ = "DELETE"
	_ = "GET"
	_ = "HEAD"
	_ = "OPTIONS"
	_ = "PATCH"
	_ = "POST"
	_ = "PUT"
	_ = "TRACE"
)

func _() {
	_, _ = http.NewRequest("CONNECT", "", http.NoBody) // want `"CONNECT" can be replaced by http\.MethodConnect`
	_, _ = http.NewRequest("DELETE", "", http.NoBody)  // want `"DELETE" can be replaced by http\.MethodDelete`
	_, _ = http.NewRequest("GET", "", http.NoBody)     // want `"GET" can be replaced by http\.MethodGet`
	_, _ = http.NewRequest("HEAD", "", http.NoBody)    // want `"HEAD" can be replaced by http\.MethodHead`
	_, _ = http.NewRequest("OPTIONS", "", http.NoBody) // want `"OPTIONS" can be replaced by http\.MethodOptions`
	_, _ = http.NewRequest("PATCH", "", http.NoBody)   // want `"PATCH" can be replaced by http\.MethodPatch`
	_, _ = http.NewRequest("POST", "", http.NoBody)    // want `"POST" can be replaced by http\.MethodPost`
	_, _ = http.NewRequest("PUT", "", http.NoBody)     // want `"PUT" can be replaced by http\.MethodPut`
	_, _ = http.NewRequest("TRACE", "", http.NoBody)   // want `"TRACE" can be replaced by http\.MethodTrace`
}

func _() {
	_, _ = http.NewRequestWithContext(nil, "CONNECT", "", http.NoBody) // want `"CONNECT" can be replaced by http\.MethodConnect`
	_, _ = http.NewRequestWithContext(nil, "DELETE", "", http.NoBody)  // want `"DELETE" can be replaced by http\.MethodDelete`
	_, _ = http.NewRequestWithContext(nil, "GET", "", http.NoBody)     // want `"GET" can be replaced by http\.MethodGet`
	_, _ = http.NewRequestWithContext(nil, "HEAD", "", http.NoBody)    // want `"HEAD" can be replaced by http\.MethodHead`
	_, _ = http.NewRequestWithContext(nil, "OPTIONS", "", http.NoBody) // want `"OPTIONS" can be replaced by http\.MethodOptions`
	_, _ = http.NewRequestWithContext(nil, "PATCH", "", http.NoBody)   // want `"PATCH" can be replaced by http\.MethodPatch`
	_, _ = http.NewRequestWithContext(nil, "POST", "", http.NoBody)    // want `"POST" can be replaced by http\.MethodPost`
	_, _ = http.NewRequestWithContext(nil, "PUT", "", http.NoBody)     // want `"PUT" can be replaced by http\.MethodPut`
	_, _ = http.NewRequestWithContext(nil, "TRACE", "", http.NoBody)   // want `"TRACE" can be replaced by http\.MethodTrace`
}

func _() {
	_ = &http.Request{
		Method: "CONNECT", // want `"CONNECT" can be replaced by http\.MethodConnect`
	}
	_ = &http.Request{
		Method: "DELETE", // want `"DELETE" can be replaced by http\.MethodDelete`
	}
	_ = &http.Request{
		Method: "GET", // want `"GET" can be replaced by http\.MethodGet`
	}
	_ = &http.Request{
		Method: "HEAD", // want `"HEAD" can be replaced by http\.MethodHead`
	}
	_ = &http.Request{
		Method: "OPTIONS", // want `"OPTIONS" can be replaced by http\.MethodOptions`
	}
	_ = &http.Request{
		Method: "PATCH", // want `"PATCH" can be replaced by http\.MethodPatch`
	}
	_ = &http.Request{
		Method: "POST", // want `"POST" can be replaced by http\.MethodPost`
	}
	_ = &http.Request{
		Method: "PUT", // want `"PUT" can be replaced by http\.MethodPut`
	}
	_ = &http.Request{
		Method: "TRACE", // want `"TRACE" can be replaced by http\.MethodTrace`
	}
}
