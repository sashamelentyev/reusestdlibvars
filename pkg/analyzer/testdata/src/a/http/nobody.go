// Code generated by usestdlibvars, DO NOT EDIT.

package http_test

import "net/http"

func _() {
	_, _ = http.NewRequest(http.MethodGet, "", nil) // want `"nil" can be replaced by http\.NoBody`
}

func _() {
	_, _ = http.NewRequestWithContext(nil, http.MethodGet, "", nil) // want `"nil" can be replaced by http\.NoBody`
}
