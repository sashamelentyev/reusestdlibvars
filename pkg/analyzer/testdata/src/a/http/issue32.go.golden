// Code generated by usestdlibvars, DO NOT EDIT.

package http_test

import "net/http"

var (
	_ = "connect"
	_ = "delete"
	_ = "get"
	_ = "head"
	_ = "options"
	_ = "patch"
	_ = "post"
	_ = "put"
	_ = "trace"
)

const (
	_ = "connect"
	_ = "delete"
	_ = "get"
	_ = "head"
	_ = "options"
	_ = "patch"
	_ = "post"
	_ = "put"
	_ = "trace"
)

var (
	_ = http.Request{Method: http.MethodConnect} // want `"connect" can be replaced by http\.MethodConnect`
	_ = http.Request{Method: http.MethodDelete} // want `"delete" can be replaced by http\.MethodDelete`
	_ = http.Request{Method: http.MethodGet} // want `"get" can be replaced by http\.MethodGet`
	_ = http.Request{Method: http.MethodHead} // want `"head" can be replaced by http\.MethodHead`
	_ = http.Request{Method: http.MethodOptions} // want `"options" can be replaced by http\.MethodOptions`
	_ = http.Request{Method: http.MethodPatch} // want `"patch" can be replaced by http\.MethodPatch`
	_ = http.Request{Method: http.MethodPost} // want `"post" can be replaced by http\.MethodPost`
	_ = http.Request{Method: http.MethodPut} // want `"put" can be replaced by http\.MethodPut`
	_ = http.Request{Method: http.MethodTrace} // want `"trace" can be replaced by http\.MethodTrace`
)

var (
	_ = &http.Request{Method: http.MethodConnect} // want `"connect" can be replaced by http\.MethodConnect`
	_ = &http.Request{Method: http.MethodDelete} // want `"delete" can be replaced by http\.MethodDelete`
	_ = &http.Request{Method: http.MethodGet} // want `"get" can be replaced by http\.MethodGet`
	_ = &http.Request{Method: http.MethodHead} // want `"head" can be replaced by http\.MethodHead`
	_ = &http.Request{Method: http.MethodOptions} // want `"options" can be replaced by http\.MethodOptions`
	_ = &http.Request{Method: http.MethodPatch} // want `"patch" can be replaced by http\.MethodPatch`
	_ = &http.Request{Method: http.MethodPost} // want `"post" can be replaced by http\.MethodPost`
	_ = &http.Request{Method: http.MethodPut} // want `"put" can be replaced by http\.MethodPut`
	_ = &http.Request{Method: http.MethodTrace} // want `"trace" can be replaced by http\.MethodTrace`
)

var (
	_, _ = http.NewRequest(http.MethodConnect, "", http.NoBody) // want `"connect" can be replaced by http\.MethodConnect`
	_, _ = http.NewRequest(http.MethodDelete, "", http.NoBody) // want `"delete" can be replaced by http\.MethodDelete`
	_, _ = http.NewRequest(http.MethodGet, "", http.NoBody) // want `"get" can be replaced by http\.MethodGet`
	_, _ = http.NewRequest(http.MethodHead, "", http.NoBody) // want `"head" can be replaced by http\.MethodHead`
	_, _ = http.NewRequest(http.MethodOptions, "", http.NoBody) // want `"options" can be replaced by http\.MethodOptions`
	_, _ = http.NewRequest(http.MethodPatch, "", http.NoBody) // want `"patch" can be replaced by http\.MethodPatch`
	_, _ = http.NewRequest(http.MethodPost, "", http.NoBody) // want `"post" can be replaced by http\.MethodPost`
	_, _ = http.NewRequest(http.MethodPut, "", http.NoBody) // want `"put" can be replaced by http\.MethodPut`
	_, _ = http.NewRequest(http.MethodTrace, "", http.NoBody) // want `"trace" can be replaced by http\.MethodTrace`
)

var (
	_, _ = http.NewRequestWithContext(nil, http.MethodConnect, "", http.NoBody) // want `"connect" can be replaced by http\.MethodConnect`
	_, _ = http.NewRequestWithContext(nil, http.MethodDelete, "", http.NoBody) // want `"delete" can be replaced by http\.MethodDelete`
	_, _ = http.NewRequestWithContext(nil, http.MethodGet, "", http.NoBody) // want `"get" can be replaced by http\.MethodGet`
	_, _ = http.NewRequestWithContext(nil, http.MethodHead, "", http.NoBody) // want `"head" can be replaced by http\.MethodHead`
	_, _ = http.NewRequestWithContext(nil, http.MethodOptions, "", http.NoBody) // want `"options" can be replaced by http\.MethodOptions`
	_, _ = http.NewRequestWithContext(nil, http.MethodPatch, "", http.NoBody) // want `"patch" can be replaced by http\.MethodPatch`
	_, _ = http.NewRequestWithContext(nil, http.MethodPost, "", http.NoBody) // want `"post" can be replaced by http\.MethodPost`
	_, _ = http.NewRequestWithContext(nil, http.MethodPut, "", http.NoBody) // want `"put" can be replaced by http\.MethodPut`
	_, _ = http.NewRequestWithContext(nil, http.MethodTrace, "", http.NoBody) // want `"trace" can be replaced by http\.MethodTrace`
)
